py_mmd_tools/nc_to_mmd.py

```diff
@@ -788,15 +788,16 @@ def get_dataset_citations(self, mmd_element, ncin):
                 if len(urls) <= i:
                     # Issue warning
                     self.missing_attributes['warnings'].append(
-                        '%s attribute is missing' % acdd_url)
+                        '%s attribute is missing' % acdd_url_key)
                     url = ''
                 else:
                     url = urls[i]
                 # Validate the url
                 if not valid_url(url):
-                    # Issue warning
-                    self.missing_attributes['warnings'].append(
-                        '"%s" in %s attribute is not a valid url' % (url, acdd_url))
+                    if url != '':
```

Comment:
Why this extra  empty url check?Whether the url here is not mandatory?

Reply:
Correct. It's not mandatory, and `validate_url` will return false for an empty string although that should be ok. So, the warning should just be issued if the provided non-empty string is not a valid url.

Reply:
Ok.